src/components/modals/ButtonsModal.tsx

```diff
@@ -33,7 +35,7 @@ export interface ButtonInfo {
 export function ButtonsModal<
   Buttons extends { [key: string]: ButtonInfo }
 >(props: {
-  bridge: AirshipBridge<keyof Buttons | undefined>
+  bridge: AirshipBridge<Array<keyof Buttons>>
```

Comment:
This is going to cause problems in edge-react-gui. One of our goals is to keep the common components in-line with each other, so we can move code between the two repos pretty easily. The `ButtonsModal` is used in *tons* of places, so changing its API like this is pretty harsh. We also lose the type-checking on these string constants.

The buttons modal was never supposed to be an all-in-one GUI toolkit. It think you could either create an `CheckboxModal` for this, or maybe even hard-code a `RequestPermissionsModal` with the functionality you need.